flexget/plugins/metainfo/nfo_lookup.py

```diff
@@ -0,0 +1,241 @@
+from __future__ import unicode_literals, division, absolute_import
+from builtins import *  # pylint: disable=unused-import, redefined-builtin
+
+import logging
+import os
+import xml.etree.ElementTree as ET
+
+from flexget import plugin
+from flexget.event import event
+from flexget.plugins.metainfo.imdb_lookup import ImdbLookup
+# from flexget.utils.imdb import make_url as make_imdb_url
+# from flexget.utils.imdb import ImdbSearch
+
+log = logging.getLogger('nfo_lookup')
+
+
+class NfoLookup(object):
+    """
+    Retrieves information from a local '.nfo' info file.
+
+    The nfo file is used by Kodi.
+
+    Example:
+        nfo_lookup: yes
+
+    WARNING: This plugin will read a file with extension '.nfo' and the same
+    name as the entry filename as an XML file using xml.etree.ElementTree from
+    the standard python library. As such, it is vulnerable to XML
+    vulnerabilities described in the link below
+    https://docs.python.org/3/library/xml.html#xml-vulnerabilities
+
+    Use this only with nfo files you have created yourself.
+    """
+    # xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
+    # Fields that will be read from the nfo file. The corresponding field in
+    # the entry will be nfo_{0} for each field name in these two lists.
+
+    # Fields that appear only once in the nfo file
+    single_value_fields = ["title", "originaltitle", "sorttitle", "rating",
+                           "year", "votes", "plot", "runtime",
+                           "id", "filenameandpath", "trailer",
+                           # thumb has multiple values, but we will get the first one
+                           "thumb"]
+    # Fields that can appear multiple times in the nfo file
+    multiple_value_fields = [
+        # "thumb",
+        "genre", "director", "actor"]
+    # xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
+
+    schema = {'type': 'boolean'}
+
+    # Run before the imdb_lookup plugin
+    @plugin.priority(150)
+    def on_task_metainfo(self, task, config):
+        # check if disabled (value set to false)
+        if not config:
+            # Config was set to 'no' instead of yes. Don't do anything then.
+            return
+
+        for entry in task.entries:
+            # If this entry was obtained from the filesystem plugin it should
+            # have a filename field. If it does not have one then there is
+            # nothing we can do in this plugin.
+            filename = entry.get('filename')
+            location = entry.get('location')
+
+            # If there is no 'filename' field there is also no nfo file
+            if filename is None or location is None:
+                log.warning("Entry '{0}' didn't come from the filesystem plugin".format(entry.get('title')))
```

Comment:
The reason I'm not returning here is that I'm currently using this plugin as a
replacement to imdb_lookup and thus if I returned here there would be no
lookup.